app/models.py

```diff
@@ -1643,6 +1643,57 @@ def serialize(self):
 
         return serialized
 
+    def serialize_with_cost_info(self):
+        serialized = self.serialize()
+        serialized["cost_details"] = {}
+        serialized["cost_in_pounds"] = "0.00"
+        serialized["is_cost_data_ready"] = True
+
+        if self.notification_type == "sms":
+            return self._get_cost_info_for_sms(serialized)
+        elif self.notification_type == "letter":
+            return self._get_cost_info_for_letter(serialized)
+
+        return serialized
+
+    def _get_cost_info_for_sms(self, serialized):
+        serialized["cost_details"]["sms_fragments"] = self.billable_units
+        serialized["cost_details"]["rate_multiplier"] = self.rate_multiplier
+        sms_rate = self._get_sms_rate()
+        serialized["cost_details"]["rate"] = str(sms_rate)
+        serialized["cost_in_pounds"] = str(self.billable_units * self.rate_multiplier * sms_rate)
+
+        return serialized
+
+    def _get_cost_info_for_letter(self, serialized):
```

Comment:
Templated letters start off with the `billable_units` set to 0 until the `create-pdf-for-templated-letter` task in template preview has been called and then has called `update-billable-units-for-letter` in api. So the cost data will not be ready immediately for any letters, not just precompiled ones.

Reply:
ahh you are right - I will update the code to reflect this 👍🏼

Reply:
Done ☺️ . Together with an extended commit message to document it ☺️